### ISSUE TITLE
Barbarian hacker intro bug

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#47]
Findings: When the disk is dragged in *after* the disk & hand logo has shown up, the intro is scrambled: 

<img width="1032" alt="Screenshot 2019-06-09 at 14 18 54" src="https://user-images.githubusercontent.com/12561945/59161060-cfc96a80-8add-11e9-90bf-d38d464d8d9d.png">

DMA debugger view: 

<img width="1032" alt="Screenshot 2019-06-09 at 16 46 22" src="https://user-images.githubusercontent.com/12561945/59161077-161ec980-8ade-11e9-83f2-d3198b49d6c2.png">

When the disk is inserted right from the start, it looks like this (red stripes are debug markers) : 

<img width="1032" alt="Screenshot 2019-06-09 at 17 38 32" src="https://user-images.githubusercontent.com/12561945/59161064-ebcd0c00-8add-11e9-837f-5a5b20292841.png">

Don't know what's going on here. Obviously, it must have something to do with the display window. Up to now, I thought the display window can only be modified by writing into DIWSTRT and DIWSTOP... 🤔

**Comment by dirkwhoffmann** [comment]
When writing to strobe register COPJMP1, Copper jumped to table 2 🤭. So mean of him.

But as always: For each captured bug, another one takes its place. 

<img width="1032" alt="Screenshot 2019-06-09 at 19 24 54" src="https://user-images.githubusercontent.com/12561945/59162181-f347e180-8aec-11e9-8192-4a43ed3d4f63.png">

The left part of the moving text shouldn't be there and the first line is scrambled.

**Comment by dirkwhoffmann** [comment]
It's a little better now, but not perfect: 

<img width="1032" alt="Screenshot 2019-06-10 at 11 12 46" src="https://user-images.githubusercontent.com/12561945/59185699-6135ec80-8b71-11e9-90cf-2b8703b8c66b.png">

The problem here is that the intro modifies registers DDFSTRT, DDFSTOP, and BPLxPTR in the middle of a rasterline when DMA is still on. Hence, register timing is crucial here. I'm going to postpone this bug until the emulator is more mature (register timing is higher level sorcery 🧟‍♀️).

**Comment by dirkwhoffmann** [comment]
😯 I screwed it up completely. Some severe bug has been introduced lately. The animation is also gone.

<img width="893" alt="Bildschirmfoto 2019-07-20 um 15 36 21" src="https://user-images.githubusercontent.com/12561945/61579363-55d8c800-ab04-11e9-90c4-6a3795b27181.png">

**Comment by dirkwhoffmann** [comment]
Animation stopped with the introduction of function skipBPLxPT(). 

skipBPLxPT() is not the culprit though. Due to another bug, writes to BPLPT registers get falsely delayed to those cycles that are skipped by skipBPLxPT(). Hence, the other bug needs to be fixed (wherever it hides).

**Comment by dirkwhoffmann** [comment]
Findings: 

<img width="980" alt="Bildschirmfoto 2019-07-21 um 08 44 40" src="https://user-images.githubusercontent.com/12561945/61588826-ef09ec00-aba1-11e9-863f-b8e840be63e7.png">

- MOVE $0018, DFFSTRT happens at cycle $16.
   This is too late to let the new value take effect in the current raster line. Therefore, DDFSTRT is still at $38 and there is enough time to set the BPLPT registers outside the DMA area. 

- In vAmiga, DMA is already on when the BPLPT registers are written.

**Comment by mithrendal** [comment]
This is better than rocket sience.

**Comment by dirkwhoffmann** [comment]
First time it really works 😎.

<img width="849" alt="Bildschirmfoto 2019-07-21 um 13 14 48" src="https://user-images.githubusercontent.com/12561945/61590447-c7268280-abb9-11e9-9a4b-6ffe2942cebf.png">

On the negative side, I'm pretty sure that my code changes break some of the DDF tests that passed before. 

Anyway, the good news is that I understand DDFSTRT / DDFSTOP a little bit better now.

**Comment by dirkwhoffmann** [comment]
Oh no. It's broken again 🙈

<img width="853" alt="Bildschirmfoto 2019-08-20 um 17 08 23" src="https://user-images.githubusercontent.com/12561945/63359667-6b444a80-c36d-11e9-8205-c936f2cfc261.png">

**Comment by mithrendal** [comment]
Is it maybe the slow blitter which is still in its infancy ?

**Comment by dirkwhoffmann** [comment]
> Is it maybe the slow blitter which is still in its infancy ?

Figured it out. It's because I changed the format of the internal DMA event table. 

To make the slow Blitter work, I have to emulate wait-states inside the CPU. To do that, I need to

1. change the IRQ handling
2. trigger the hsync handler at the beginning of the next line and not at the end of the current. 
3. modify the value of pos.h when it wraps over

Task 1. has been solved and brought the Musashi IRQ issue to surface (which is good).
Task 2. and 3. sound simple, but they aren't. All DMA stuff is tightly glued together like in a real Amiga. One sub-task is to change format of the internal DMA event table which caused the bug.
